plugins/postcss-logical/src/lib/clone-declaration.ts

```diff
@@ -4,6 +4,12 @@ export function cloneDeclaration(
 	declaration: Declaration,
 	value: string,
 	prop: string,
-) : Declaration {
-	return declaration.cloneBefore({ value, prop });
+): void {
+	if (declaration.parent && declaration.parent.some((x) => {
+		return x.type == 'decl' && x.prop === prop && x.value === value;
+	})) {
+		return;
+	}
```

Comment:
I am unsure about this.
It is a nice addition that prevents duplicate declarations like those added in `css-has-pseudo`.

But it can also slow down the plugin.
I don't mind removing this again.

Reply:
I don't think speed would be much impacted in the average CSS file, let's keep this :)